pyDrop/clustering/PCRData.py

```diff
@@ -0,0 +1,444 @@
+"""
+=========================================================================================
+                    PCRData: Data Handling and Clustering Pipeline
+=========================================================================================
+PCRData: class wrapper for most of the plotting and clustering functionality in pyDrop
+"""
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import re
+from sklearn import cluster
+from sklearn.metrics.cluster import rand_score, homogeneity_score, completeness_score, v_measure_score, silhouette_score
+
+import os
+import math
+from scipy import stats
+from scipy.stats import poisson, norm, chi2, cosine, exponnorm, hypsecant, logistic, genlogistic, dweibull, cauchy, exponweib
+
+from pyDrop.exceptions import (PCRDataReadingException,
+                               FeaturesImproperlySpecified,
+                               LabelsImproperlySpecified,
+                               TooManyDimensions,
+                               SupervisedDataRequired,
+                               NumClustersModelValueError,
+                               ImproperClusterModel)
+
+class PCRData:
+    """Loads Data from file or pandas dataframe. Includes options 
+    for column naming which is required if input data does not follow
+    the format ch1, ch2, ... cluster_* (case in-sensitive).
+    Parameters
+    ----------
+    data : pandas.dfame | str
+        pandas dataframe or file path to csv file containing 
+    X_columns : list(str)
```

Comment:
comment: lower vs upper isn't a huge deal, but the `x_column` and `y_column` should at least follow the same convention IMO